hirosystems/stacks-blockchain-api/templates/api-writer/statefulset.yaml

```diff
@@ -212,47 +214,52 @@ spec:
             - name: PG_PORT
               value: {{ include "postgresql.service.port" .Subcharts.postgresql | quote }}
             - name: PG_USER
               value: {{ include "stacksBlockchainApi.postgresql.username" . }}
-            - name: PG_DATABASE
-              value: {{ include "postgresql.database" .Subcharts.postgresql | quote }}
-            - name: PG_SCHEMA
-              value: {{ include "postgresql.database" .Subcharts.postgresql }}
-            - name: PG_CONNECTION_POOL_MAX
-              value: {{ .Values.apiWriter.config.pgConnectionPoolMax | quote }}
             - name: PG_PASSWORD
               valueFrom:
                 secretKeyRef:
                   key: {{ include "postgresql.adminPasswordKey" .Subcharts.postgresql | quote }}
                   name: {{ include "postgresql.secretName" .Subcharts.postgresql | quote }}
-            - name: PG_IDENT_INDEX_TYPE
-              value: hash
-            - name: NODE_ENV
-              value: production
-            - name: BNS_IMPORT_DIR
-              value: {{ .Values.apiWriter.persistence.bns.mountPath }}
-            - name: STACKS_API_EVENT_REPLAY_LOG_LEVEL
-              value: debug
-            - name: STACKS_API_ENABLE_FT_METADATA
-              value: {{ ternary "1" "0" .Values.apiWriter.config.enableFtMetadata | quote }}
-            - name: STACKS_API_ENABLE_NFT_METADATA
-              value: {{ ternary "1" "0" .Values.apiWriter.config.enableNftMetadata | quote }}
-          command:
-            - sh
-            - -c
-            {{- if .Values.apiWriter.config.replayEvents.prunedMode }}
-            - |
-              node ./lib/index.js import-events --file ${STACKS_EXPORT_EVENTS_FILE} --wipe-db --force --mode pruned
-            {{- else }}
-            - |
-              node ./lib/index.js import-events --file ${STACKS_EXPORT_EVENTS_FILE} --wipe-db --force
+            - name: PG_APPLICATION_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: PG_CONNECTION_POOL_MAX
+              value: {{ default "50" .Values.apiWriter.config.pgConnectionPoolMax | quote }}
+            {{- if .Values.apiWriter.extraEnvVars }}
+            {{- include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVars "context" $) | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- if .Values.apiWriter.extraEnvVarsCM }}
+            - configMapRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVarsCM "context" $) }}
+            {{- end }}
+            {{- if .Values.apiWriter.extraEnvVarsSecret }}
+            - secretRef:
+                name: {{ include "common.tplvalues.render" (dict "value" .Values.apiWriter.extraEnvVarsSecret "context" $) }}
             {{- end }}
+          command: node ./lib/index.js from-parquet-events --workers={{ .Values.apiWriter.config.replayEvents.workers }}
```

Comment:
After the import is complete, does the container exit gracefully? Or does it switch over to running the API like normal?

Reply:
yes, it does exit gracefully and switches over to the main container which essentially runs the same image tag.
here is a log output
```
{"level":"info","time":"2023-08-25T19:54:13.604Z","pid":7,"hostname":"mainnet-api-replay-writer-0","name":"stacks-blockchain-api","component":"event-replay","msg":"RAW events process started"}
{"level":"info","time":"2023-08-25T19:54:13.605Z","pid":7,"hostname":"mainnet-api-replay-writer-0","name":"stacks-blockchain-api","component":"event-replay","msg":"Generating ID files for 4 workers"}
{"level":"error","time":"2023-08-25T19:54:13.689Z","pid":7,"hostname":"mainnet-api-replay-writer-0","name":"stacks-blockchain-api","component":"core-api","msg":"Received uncaughtException: Error: IO Error: No files found that match the pattern \"/hirosystems/api-writer/data/events/attachments/new/canonical/*.parquet\""}
{"level":"error","time":"2023-08-25T19:54:13.689Z","pid":7,"hostname":"mainnet-api-replay-writer-0","name":"stacks-blockchain-api","component":"core-api","msg":"Shutting down... received uncaughtException."}
{"level":"info","time":"2023-08-25T19:54:13.689Z","pid":7,"hostname":"mainnet-api-replay-writer-0","name":"stacks-blockchain-api","component":"core-api","msg":"App shutdown successful."}```